lib/arel/middleware/chain.rb

```diff
@@ -21,9 +21,9 @@ def execute(sql, binds = [], &execute_sql)
         check_middleware_recursion(sql)
 
         updated_context = context.merge(original_sql: sql)
-        enhanced_arel = Arel.enhance(Arel.sql_to_arel(sql, binds: binds))
+        arel = Arel.sql_to_arel(sql, binds: binds)
```

Comment:
As I like to use the SQL (and binds) as the cache key, making this sql_to_arel part of the executor and make it part of the cached value will save some unnecessary operations. 

However, I'm not sure how to do this without changing the DSL of the middleware.

Reply:
How I'm thinking about it, maybe having a middleware that does the caching isn't the easiest (mostly implementation wise) solution. Instead, I could add some first-class caching support to the chain that just wraps all this execution stuff.

Reply:
and that's what I did

Reply:
Awesome!